assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js

```diff
@@ -61,10 +61,16 @@ export default function PublicationApprovedOverlayNotification() {
 			( select ) =>
 				select( MODULES_READER_REVENUE_MANAGER ).getSettings() || {}
 		);
-	const initialPublicationOnboardingStateChanged = useRef(
-		publicationOnboardingStateChanged
+
+	// Check if settings has been resolved.
+	const hasSettingsResolved = useSelect( ( select ) =>
+		select( MODULES_READER_REVENUE_MANAGER ).hasFinishedResolution(
+			'getSettings'
+		)
 	);
 
+	const initialPublicationOnboardingStateChanged = useRef( undefined );
```

Comment:
Is it necessary to specify `undefined` here? The initial value of `useRef` is `undefined` by default.

```suggestion
	const initialPublicationOnboardingStateChanged = useRef();
```